pt/branch.c

```diff
@@ -0,0 +1,105 @@
+// SPDX-License-Identifier: GPL-2.0-only
+// Copyright (c) 2022 Intel Corporation.
+
+#include "utils.h"
+
+/**
+ * disable branch check :
+ *	disbale branch and check package
+ *	TIP/FUP/TNT: should not be expected
+ *	check basic expected packet inside traces
+ *	PASS will return 0 and FAIL will return 1
+ */
+int disablebranchcheck(void)
+{
+	unsigned int cond = 0, stp = 0, END = 0xFFFF, FAIL = 0, result = 0;
+	struct perf_event_attr attr;
+	struct perf_event_mmap_page *pmp;
+	int fde, fdi;
+	long bufSz;
+	uint64_t **bufM, head;
+	enum pt_packet_type c_type;
+	char *en_trace = "en.trace";
+	char *de_trace = "de.trace";
+	pid_t pid;
+	//	Set buffersize as 2 pagesize
+	bufSz = 2 * PAGESIZE;
+	// initial attribute for PT
+	iniEvtAttr(&attr);
+	// set bit 0 and unset bit 13 for disabling branch
+	attr.config &= ~(1 << 13);
+	attr.config |= 1;
+
+	pid = getpid();
+	// only get trace for own pid
+	fde = sys_perf_event_open(&attr, pid, -1, -1, 0);
+	if (fde < 0) {
+		perror("perf_event_open");
+		FAIL = 1;
+		goto onerror;
+	}
+	/* map event : full */
+
+	bufM = creaMap(fde, bufSz, 0, &fdi);
+
+	if (!bufM || (bufM)[0] == MAP_FAILED || (bufM)[1] == MAP_FAILED) {
+		perror("Full Trace creaMap");
+
+		close(fde);
+		FAIL = 1;
+		goto onerror;
+	}
+
+	ioctl(fde, PERF_EVENT_IOC_RESET);
+
+	ioctl(fde, PERF_EVENT_IOC_ENABLE);
+
+	/* stop tracing */
+	ioctl(fde, PERF_EVENT_IOC_DISABLE);
+
+	printf("bufSz = %ld\n", bufSz);
+	pmp = bufM[0];
+	head = (*(volatile uint64_t *)&(pmp->aux_head));
+	printf("head = %ld\n", head);
+
+	c_type = ppt_fup;
+	if (seekPck_w_lib(c_type, bufM[1], head) == 0) {
+		printf("FUP should not be found!\n");
+		FAIL = 1;
+	}
+	c_type = ppt_tip;
+	if (seekPck_w_lib(c_type, bufM[1], head) == 0) {
+		printf("TIP should not be found!\n");
+		FAIL = 1;
+	}
+	c_type = ppt_tnt_64;
+	if (seekPck_w_lib(c_type, bufM[1], head) == 0) {
+		printf("TNT should not be found!\n");
+		FAIL = 1;
+	}
+
+	/* unmap and close */
+	delMap(bufM, bufSz, 1, fdi);
+	close(fde);
+onerror:
+	printf("USER trace address check %s\n", FAIL ? "FAIL" : "PASS");
```

Comment:
From my record, we have a rule for the results' info print. e.g.
[PASS]case_name: comments
[FAIL]case_name: comments